### PR TITLE
[SELC-5097] Feat: Added function for CreateOnboardingRequest

### PR DESCRIPTION
**Pull request by flaminiaScarciofolo**
#### List of Changes
- Added function for CreateOnboardingRequest

#### Motivation and Context
We need a function to persist onboarding for aggregates

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/service/CompletionServiceDefault.java [hunk 1]
```diff
@@ -299,4 +306,13 @@ public void persistActivatedAt(Onboarding onboarding) {
                 .update("activatedAt = ?1 and updatedAt = ?2 ", now, now)
                 .where("_id", onboarding.getId());
     }
+
+    @Override
+    public Onboarding createAggregateOnboardingRequest(OnboardingAggregateOrchestratorInput onboardingAggregateOrchestratorInput) {
+        Onboarding onboardingToUpdate = onboardingMapper.mapToOnboarding(onboardingAggregateOrchestratorInput);
+        onboardingToUpdate.setWorkflowType(CONFIRMATION_AGGREGATE);
+        onboardingToUpdate.setStatus(OnboardingStatus.PENDING);
+        onboardingRepository.persistOrUpdate(onboardingToUpdate);
```
Comment (manuraf) [inline apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/service/CompletionServiceDefault.java:315]: pay attention that we should create a new onboarding record....it is important to set a new objectid for persisting a new object

File: apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/mapper/OnboardingMapper.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import org.mapstruct.Mapping;
+import org.mapstruct.Named;
+
+import java.util.Objects;
+
+@Mapper(componentModel = "cdi")
+public interface OnboardingMapper {
+
+    @Mapping(target = "institution", expression = "java(mapInstitution(input.getAggregate(), input.getInstitution()))")
+    @Mapping(target = "aggregator", expression = "java(mapAggregator(input.getInstitution()))")
```
Comment (manuraf) [inline apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/mapper/OnboardingMapper.java:17]: you could avoid to create a method for mapping aggregator....you could use `@Mapping(target = "aggregator", source = "institution")`

**Review by manuraf — CHANGES_REQUESTED** [review]

File: apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/mapper/OnboardingMapper.java (hunk 2 above)
Comment (flaminiaScarciofolo) [reply apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/mapper/OnboardingMapper.java:17]: We need to set institutionType and Origin from aggregator. I'll try to replace with @Mapping

File: apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/mapper/OnboardingMapper.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    Onboarding mapToOnboarding(OnboardingAggregateOrchestratorInput input);
+
+    @Named("mapInstitution")
+    default Institution mapInstitution(Institution aggregator, Institution institution) {
+        if (Objects.nonNull(institution)) {
+            aggregator.setOrigin(institution.getOrigin());
+            aggregator.setInstitutionType(institution.getInstitutionType());
+        }
+        return aggregator;
+    }
```
Comment (manuraf) [inline apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/mapper/OnboardingMapper.java:28]: ditto as mapperAggregator

**Review by manuraf — APPROVED** [review]

**Pull request merged**